### PR TITLE
Config file read based on ucfg

### PR DESCRIPTION
**Pull request by urso**
Use ucfg to parse configuration files and populate configuration.

Introduces support for dotted names in config files. e.g. use `output.elasticsearch.hosts: [...]` instead of indents.

based on #973 adding ucfg to beats

### CODE REVIEW

File: libbeat/common/datetime_test.go [hunk 1]
```diff
@@ -28,20 +28,23 @@ func TestParseTime(t *testing.T) {
...
-			Input:  "2015-03-31T11:19:05.112Z",
-			Output: time.Date(2015, time.March, 31, 11, 19, 05, 112*1e6, time.UTC),
-		},
-		{
-			Input:  "2015-04-31T11:19:05.112Z",
-			Output: time.Date(2015, time.April, 31, 11, 19, 05, 112*1e6, time.UTC),
-		},
+
+		/*
+			// Golang time pkg happily parses 'wrong' dates like these.
```
Comment (ruflin) [inline libbeat/common/datetime_test.go]: Did you keep that commented out on purpose?

File: libbeat/common/datetime_test.go [hunk 2]
```diff
@@ -28,20 +28,23 @@ func TestParseTime(t *testing.T) {
...
+			},
+			{
+				Input:  "2015-03-31T11:19:05.112Z",
+				Output: time.Date(2015, time.March, 31, 11, 19, 05, 112*1e6, time.UTC),
+			},
+			{
+				Input:  "2015-04-31T11:19:05.112Z",
+				Output: time.Date(2015, time.April, 31, 11, 19, 05, 112*1e6, time.UTC),
+			},
+		*/
```
Comment (urso) [inline libbeat/common/datetime_test.go]: this has been fixed in go 1.6 .

File: libbeat/common/datetime_test.go (hunk 1 above)
Comment (urso) [inline libbeat/common/datetime_test.go]: we can remove or keep in.

It's basically testing some standard library oddities that have been fixed on go 1.6

File: filebeat/config/config_test.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ func TestReadConfig(t *testing.T) {
 	assert.Equal(t, "/var/log/s*.log", prospectors[0].Paths[1])
 	assert.Equal(t, "log", prospectors[0].Input)
 	assert.Equal(t, 3, len(prospectors[0].Harvester.Fields))
-	assert.Equal(t, "1", prospectors[0].Harvester.Fields["review"])
+	assert.Equal(t, int64(1), prospectors[0].Harvester.Fields["review"])
```
Comment (urso) [inline filebeat/config/config_test.go]: Note: fields are parsed as is and types are preserved when encoding to json. This is a change from old 'fields' support in filebeat.

**Comment by urso** [comment]
jenkins failure some redis docker image problem. Another run: http://build-eu-00.elastic.co/view/Beats/job/Beats-PR/744/

File: libbeat/cfgfile/cfgfile.go [hunk 4]
```diff
@@ -53,13 +55,17 @@ func Read(out interface{}, path string) error {
 	if err != nil {
 		return fmt.Errorf("Failed to read %s: %v. Exiting.", path, err)
 	}
-
 	filecontent = expandEnv(filecontent)
 
-	if err = yaml.Unmarshal(filecontent, out); err != nil {
-		return fmt.Errorf("YAML config parsing failed on %s: %v. Exiting", path, err)
+	config, err := yaml.NewConfig(filecontent, ucfg.PathSep("."))
```
Comment (ruflin) [inline libbeat/cfgfile/cfgfile.go]: Is it necessary to define PathSep twice? Line 65

**Comment by ruflin** [comment]
LGTM

File: libbeat/cfgfile/cfgfile.go (hunk 4 above)
Comment (urso) [inline libbeat/cfgfile/cfgfile.go]: PathSep is some option to be passed to Merge and Unpack. Without this option PathSep is not applied. Due to Configs being reusable and potentially in multiple configs as Sub-configs, options are not stored within the configuration for consistency reasons.

**Pull request merged**